src/file.ts

```diff
@@ -497,8 +497,6 @@ class File extends ServiceObject<File> {
    * const file = myBucket.file('my-file');
    */
   constructor(bucket: Bucket, name: string, options: FileOptions = {}) {
-    name = name.replace(/^\/+/, '');
```

Comment:
@callmehiphop since your question hasn't been answered yet I will try.  I think this library is trying to prevent people from having double slashes (https://github.com/googleapis/google-cloud-node/issues/1972).  My original PR was backwards compatible (and awkward), but I agreed with @stephenplusplus that it would be cleaner to undo removing leading slashes and let the developer choose the path.

Other google libraries like python do not change files: https://cloud.google.com/python/
So, there are no guard rails to prevent Python code like:
```python
from google.cloud import storage
#...
storage_client = storage.Client()
bucket = storage_client.get_bucket(bucket_name)
blob = bucket.blob("/path/to/image.png")
```

If I have written to storage with Python with a leading slash it is not readable without the workaround from my original commit notes.  Additionally, the `getFiles()` is also incorrect, so I am unable to find existing files (I would need to guess how many leading slashes there are?).